File: quiz/deno/src/statemaschine/statemaschine.ts
import { InitState } from '@states/init-state.ts';
import { IState } from '@states/state.ts';
import { UI } from '@src/utils.ts';

export class StateMaschine {
	private state: IState = new InitState();

	public async start() {
		while (this.state) {
			UI.clear();
			this.state = await this.state.run();
		}
	}
}

File: quiz/go/statemaschine/statemaschine.go
package statemaschine

import (
	"github.com/DoktorLenz/kp-wise23-24/quiz/go/statemaschine/states"
	"github.com/DoktorLenz/kp-wise23-24/quiz/go/utils"
)

type StateMaschine struct {
	state states.IState
}

func NewStateMachine() *StateMaschine {
	return &StateMaschine{state: &states.InitState{}}
}

func (sm *StateMaschine) Start() {
	for sm.state != nil {
		utils.Clear()
		sm.state = sm.state.Run()
	}
}
